app/pipe_router.go

```diff
@@ -78,6 +86,16 @@ func NewLocalPipeRouter() PipeRouter {
 	return pipeRouter
 }
 
+func (pr *localPipeRouter) Exists(_ context.Context, id string) (bool, error) {
```

Comment:
Yes sorry should have said: in the multitenant case, I wanted to differentiate checking if something exists from staking a claim on it, which has higher overheads.